[GStreamer][MediaStream][Westeros] Enable 'immediate-output' on streaming through westerossink

eocanha opened the pull request:
#### 4861f4a2898f4b3749e81c3c23690bd86afe3da8
<pre>
[GStreamer][MediaStream][Westeros] Enable &apos;immediate-output&apos; on streaming through westerossink
<a href="https://bugs.webkit.org/show_bug.cgi?id=259930">https://bugs.webkit.org/show_bug.cgi?id=259930</a>

Reviewed by Philippe Normand.

<a href="https://bugs.webkit.org/show_bug.cgi?id=259357">https://bugs.webkit.org/show_bug.cgi?id=259357</a> removed the configuration of the &quot;immediate-output&quot; property for westeros sink, but it&apos;s still needed for mediastream content.

Original author: Vivek.A &lt;Vivek_Arumugam@comcast.com&gt;
See: <a href="https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1136">https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1136</a>

* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::configureElementPlatformQuirks): Configure the immediate-output property for westerossink.

Canonical link: <a href="https://commits.webkit.org/266720@main">https://commits.webkit.org/266720@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/3caec3765cab2a9ee3d1bb6b799e7eca635dd0db

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/14473 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/14783 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/15127 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/16218 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/13687 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/14608 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/17298 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/14860 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/16347 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/14654 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/15184 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/12289 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16942 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/12469 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/13052 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/20059 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/13546 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/13217 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/16449 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/13772 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/11604 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/13059 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3518 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/17396 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/13614 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn commented on Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
```diff
@@ -2305,6 +2305,17 @@ void MediaPlayerPrivateGStreamer::configureElementPlatformQuirks(GstElement* ele
 #endif
 #endif
 
+#if USE(WESTEROS_SINK)
```
Can we remove this ifdef?

eocanha replied:
It would mean one more runtime string comparison for all the other platforms with quirks, but the impact of that would be negligible.

philn commented on Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
```diff
@@ -2305,6 +2305,17 @@ void MediaPlayerPrivateGStreamer::configureElementPlatformQuirks(GstElement* ele
+#if ENABLE(MEDIA_STREAM)
+        if (m_streamPrivate && gstObjectHasProperty(element, "immediate-output")) {
+            GST_DEBUG_OBJECT(pipeline(), "Enable 'immediate-output' in WesterosSink");
+            g_object_set(G_OBJECT(element), "immediate-output", TRUE, nullptr);
```
No need for `G_OBJECT` cast.

eocanha committed:
[GStreamer][MediaStream][Westeros] Enable 'immediate-output' on streaming through westerossink
https://bugs.webkit.org/show_bug.cgi?id=259930

Reviewed by Philippe Normand.

https://bugs.webkit.org/show_bug.cgi?id=259357 removed the configuration of the "immediate-output" property for westeros sink, but it's still needed for mediastream content.

Original author: Vivek.A <Vivek_Arumugam@comcast.com>
See: https://github.com/WebPlatformForEmbedded/WPEWebKit/pull/1136

* Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp:
(WebCore::MediaPlayerPrivateGStreamer::configureElementPlatformQuirks): Configure the immediate-output property for westerossink.

Canonical link: https://commits.webkit.org/266720@main
